File: app/src/main/java/com/inspiringteam/xchange/di/scopes/AppScoped.java
package com.inspiringteam.xchange.di.scopes;


import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Scope;

/**
 * Replacement scope for @Singleton to improve readability
 */
@Documented
@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface AppScoped {
}

File: app/src/main/java/com/inspiringteam/xchange/di/scopes/AppScoped.kt
package com.inspiringteam.xchange.di.scopes

import javax.inject.Scope

/**
 * Replacement scope for @Singleton to improve readability
 */
@MustBeDocumented
@Scope
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class AppScoped 